src/cpp/src/speculative_decoding/speculative_decoding_impl.cpp

```diff
@@ -47,15 +47,21 @@ ContinuousBatchingPipeline::SpeculativeDecodingImpl::SpeculativeDecodingImpl(con
 
     if (is_draft_scheduler_undefined) {
         // split KV cache to 2 caches for main and draft models
-        auto compute_total_hidden_size = [] (const std::vector<KVHeadConfig>& kv_cache_config) -> size_t {
+        auto compute_total_hidden_size = [] (const std::shared_ptr<ov::Model>& model) -> size_t {
             size_t total_hidden_size = 0;
-            for (auto & config : kv_cache_config) {
-                total_hidden_size += config.k_head_size * config.num_k_heads + config.v_head_size * config.num_v_heads;
+            for (const auto& param_ptr : model->get_parameters()) {
+                const auto& name = param_ptr->get_friendly_name();
+                if (name.find("key_cache.") == 0) {
+                    auto pa_op = param_ptr->get_output_target_inputs(0).begin()->get_node();
+                    const auto& rt_info = pa_op->get_rt_info();
```

Comment:
`Node`'s rt info isn't documented, but I assume is the same thing as `Model`'s one. I got curios, is `Node`'s rt info reflected in xml if `ov::Model` is serialized ?